spec/lib/whitehall_importer/migrate_assets_spec.rb

```diff
@@ -0,0 +1,143 @@
+# frozen_string_literal: true
+
+RSpec.describe WhitehallImporter::MigrateAssets do
+  describe ".call" do
+    let(:asset_id) { "847150" }
+    let(:original_asset_url) { "https://asset-manager.gov.uk/blah/#{asset_id}/asset" }
+    let(:asset) { build(:whitehall_migration_asset_import) }
+
+    before :each do
+      stub_any_asset_manager_call
+    end
+
+    it "should take a WhitehallMigration::DocumentImport record as an argument" do
+      expect { described_class.call(build(:whitehall_migration_document_import)) }.not_to raise_error
+    end
+
+    it "should mark each asset as processing before its final state" do
+      whitehall_import = build(:whitehall_migration_document_import, assets: [asset])
+      expect(asset).to receive(:update!).with(state: "processing").once.ordered
+      expect(asset).to receive(:update!).with(hash_including(:state)).once.ordered
+      described_class.call(whitehall_import)
+    end
+
+    it "should skip migrating any assets that have already been processed" do
+      asset = create(:whitehall_migration_asset_import, state: "removed")
+      whitehall_import = build(:whitehall_migration_document_import, assets: [asset])
+      expect(asset).not_to receive(:update!)
```

Comment:
I wonder if here we should be asserting more on the side-effects. Namely that there aren't any asset manager requests?